core/GlobalState.cc

```diff
@@ -821,9 +855,31 @@ SymbolRef GlobalState::enterSymbol(Loc loc, SymbolRef owner, NameRef name, u4 fl
 
     ENFORCE(!symbolTableFrozen);
 
-    SymbolRef ret = SymbolRef(this, symbols.size());
-    store = ret; // DO NOT MOVE this assignment down. emplace_back on symbol invalidates `store`
-    symbols.emplace_back();
+    SymbolRef ret;
```

Comment:
WDYT about splitting this method into 4 and inlining it into the actual ones?

Reply:
Works for me.